flows/engine/services.go

```diff
@@ -1,23 +1,29 @@
 package engine
 
 import (
+	"github.com/nyaruka/goflow/assets"
 	"github.com/nyaruka/goflow/flows"
 )
 
 // WebhookService resolves a session to a webhook provider
 type WebhookService func(flows.Session) flows.WebhookProvider
 
+// NLUService resolves a session and classifier to an NLU provider
+type NLUService func(flows.Session, assets.Classifier) flows.NLUProvider
```

Comment:
is `Service` the right word here? This feels like a `Factory` to me in this case. IE, here's a classifier, build be a provider.

Reply:
The current "services" interface looks like this when used in actions...

```
s := engine.Services().Airtime(session)
s.Transfer(...)

engine.Services().NLU(session, asset)
s.Classify(...)

engine.Services().Webhooks(session)
s.Call(...)
```

So maybe instead of `FooProvider` it could be `FooService` and instead of `FooService` it would be `FooServiceFactory`? I'm fine with that.

Reply:
I think I like that better, `provider` vs `service` doesn't really mean anything while `service` vs `service_factory` is a bit clearer I think.

Reply:
Reworked other services in 0f446ae0413b80735ef3c9b2adb5e967416cba90 so keep this PR on theme